src/partition/repart.c

```diff
@@ -2232,40 +2256,42 @@ static int context_acquire_partition_uuids_and_labels(Context *context) {
         assert(context);
 
         LIST_FOREACH(partitions, p, context->partitions) {
-                assert(sd_id128_is_null(p->new_uuid));
+                if (sd_id128_is_null(p->new_uuid)) {
+                        /* Not explicitly set by user! */
 
-                /* Never touch foreign partitions */
-                if (PARTITION_IS_FOREIGN(p)) {
-                        p->new_uuid = p->current_uuid;
+                        /* Never touch foreign partitions */
+                        if (PARTITION_IS_FOREIGN(p)) {
+                                p->new_uuid = p->current_uuid;
 
-                        if (p->current_label) {
-                                p->new_label = strdup(p->current_label);
-                                if (!p->new_label)
-                                        return log_oom();
-                        }
+                                if (p->current_label) {
+                                        p->new_label = strdup(p->current_label);
+                                        if (!p->new_label)
+                                                return log_oom();
+                                }
 
-                        continue;
-                }
+                                continue;
+                        }
```

Comment:
this block should be left before the `if (sd_id218_is_null())…` check you added. A "foreign" partition is one that already exists and for which we have no partition definition file. We generally want to leave those untouched, hence we should have an early check that in that case we copy old to new label + uuid.